elytron/WFCORE-4496-principal-transformer-in-aggregate-realm.adoc

```diff
@@ -0,0 +1,75 @@
+= [WFCORE-4496] Adding principal-transformer in aggregate-realm
+:author:            Ashley Abdel-Sayed
+:email:             aabdelsa@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+Elytron currently has an aggregate security realm which is a combination of two or more realms: an authentication realm
+and one or more authorization realms. Currently, the principal undergoes transformations before being used to load the
+aggregate identity, but there is no way to transform the principal in between loading the authentication identity and
+loading the authorization identity. This would be needed in the case where the principal stored in the authentication realm
+is different than the principal stored in the authorization realm(s).
+
+This enhancement would allow transformations to be applied to the principal after loading the authentication identity
+and before loading the authorization identity. The principal-transformer attribute will be added to aggregate-realm and
+would be a reference to a transformer defined in the mappers configuration. It will be an optional attribute.
+
+** Example configuration:
+
+```
+   /subsystem=elytron/aggregate-realm=AggregateRealm:add(authentication-realm=KeyStoreRealm,authorization-realm=ApplicationRealm,principal-transformer=myCustomTransformer)
+```
```

Comment:
Can by in the example configuration simple mappers configuration for elytron subsystem?
I think it will be useful to have this example configuration of a simple principal transformer in the proposal.
@fjuma Do you agree?

Reply:
No this does not need a persisted example, generally we avoid that and focus on the management model.

These proposals are to enable communicate planned features before and as they are being developed - a common problem we had previously was that features were only visible to other teams once development was complete.

We also need to keep into perspective this is a proposal of planned work NOT an in-depth account of what did happen.  Of course if the implementation deviates from the plans we need to keep the plans up to date but after a certain point we do need to consider them as plans.

All enhancements also require community documentation, once that is written that is where we should be prioritising communicating configuration details as it is the community documentation and not the proposal that is used by the end users attempting to make use of the feature.

Related to this specific point we also do not include the persisted representation in the community documentation.